src/tablet/tablet_impl.cc

```diff
@@ -4849,6 +4867,16 @@ void TabletImpl::CreateProcedure(RpcController* controller, const openmldb::api:
     sp_cache_->InsertSQLProcedureCacheEntry(db_name, sp_name, sp_info_impl, session.GetCompileInfo(),
                                             batch_session.GetCompileInfo());
 
+    if (sp_info.type() == type::ProcedureType::kReqDeployment) {
+        auto collector_key = absl::StrCat(db_name, ".", sp_name);
+        auto s = deploy_collector_->AddDeploy(collector_key);
```

Comment:
AddDeploy in `RefreshTableInfo`

Reply:
how about lazy recover from `sp_cache_`, so in https://github.com/4paradigm/OpenMLDB/pull/1521/files#diff-eeaff339ba9a4857121028e73b5e1f32beff596c961a3a270cf028709c3c1ee9R5041, if it failed because of deploy key not found, will retry with `AddDeploy` first. The benefit is, as long as `sp_cache_` is corret:
- we don't even have to handle `create procedure` event
- when `deploy_stats` not enabled, there is no overhead for `create procedure`. though that overhead is low already